pkg/clients/openshift.go

```diff
@@ -288,3 +330,56 @@ func (o OpenshiftClient) SubjectRulesReview(user, namespace string, log *logging
 	}
 	return authorization.ConvertAPIPolicyRulesToRBACPolicyRules(pr), nil
 }
+
+// CovertRegistryImagesToSpecs - Return APB specs from internal OCP registry
+func (o OpenshiftClient) ConvertRegistryImagesToSpecs(log *logging.Logger, imageList []string) ([]FQImage, error) {
+	var fqList []FQImage
+	fqImage := FQImage{}
+	var err error
+	r := &ImageList{}
+	err = o.imageRestClient.Get().
+		Resource("images").
+		Do().
+		Into(r)
+
+	for _, image := range r.Items {
+		var imageName = strings.Split(image.DockerImage, "@")[0]
+		for _, providedImage := range imageList {
+			if providedImage == imageName {
+				encodedSpec := image.Metadata.ContainerConfig.Labels.Spec
+				decodedSpec, err := b64.StdEncoding.DecodeString(encodedSpec)
+				if err != nil {
+					return nil, fmt.Errorf("Failed to grab encoded spec label")
+				}
+				fqImage.Name = imageName
+				fqImage.DecodedSpec = decodedSpec
+				fqList = append(fqList, fqImage)
+			}
+		}
+	}
+	if err != nil {
+		log.Errorf("error - %v\n", err)
+		return nil, err
+	}
+
+	return fqList, nil
+}
+
+// ListRegistryImages - List images in internal OpenShift registry
+func (o OpenshiftClient) ListRegistryImages(log *logging.Logger) (images []string, err error) {
+	r := &ImageList{}
+	var imageList []string
+	err = o.imageRestClient.Get().
+		Resource("images").
+		Do().
+		Into(r)
+
+	for _, image := range r.Items {
+		imageList = append(imageList, strings.Split(image.DockerImage, "@")[0])
+	}
+	if err != nil {
```

Comment:
This is checking the error return from the `imageRestClient`? If there's an error from the client how can `r` be filled out and have items?

I think line 380 error check should be on line 377 BEFORE you loop through `r.Items`.